apps/analysis_framework/models.py

```diff
@@ -64,21 +61,92 @@ def can_modify(self, user):
         * user is super user, or
         * the framework belongs to a project where the user is admin
         """
-        import project
         return (
             self.created_by == user or
             user.is_superuser or
-            project.models.ProjectMembership.objects.filter(
-                project__in=self.project_set.all(),
+            AnalysisFrameworkMembership.objects.filter(
                 member=user,
-                role__in=project.models.ProjectRole.get_admin_roles(),
+                framework=self,
+                role__can_edit_framework=True
+            ).exists()
+        )
+
+    def can_clone(self, user):
+        return (
+            not self.is_private or
+            AnalysisFrameworkMembership.objects.filter(
+                member=user,
+                framework=self,
+                role__can_clone_framework=True,
             ).exists()
         )
 
     def get_entries_count(self):
         from entry.models import Entry
         return Entry.objects.filter(analysis_framework=self).count()
 
+    def get_or_create_owner_role(self):
+        permission_fields = self.get_owner_permissions()
+        role, created = AnalysisFrameworkRole.objects.get_or_create(
+            **permission_fields,
+            defaults={
+                'title': 'Owner Role(' + ('Private)' if self.is_private else 'Public)'),
+            }
+        )
+        return role
+
+    def get_or_create_editor_role(self):
+        permission_fields = self.get_editor_permissions()
+
+        role, created = AnalysisFrameworkRole.objects.get_or_create(
+            **permission_fields,
+            defaults={
+                'title': 'Editor Role(' + ('Private)' if self.is_private else 'Public)'),
+            }
+        )
+        return role
+
+    def get_or_create_default_role(self):
+        # For now, same for both private and public, change later if needed
+        title = 'Default(Private)' if self.is_private else 'Default(Public)'
+        role, created = AnalysisFrameworkRole.objects.get_or_create(
+            is_default_role=True,
+            defaults={
+                'can_use_in_other_projects': True,
+                'title': title,
+            }
+        )
+        return role
+
+    def add_member(self, user, role=None, added_by=None):
+        role = role or self.get_or_create_default_role()
+        return AnalysisFrameworkMembership.objects.get_or_create(
+            member=user,
+            framework=self,
+            defaults={
+                'added_by': added_by,
+                'role': role,
+            },
+        )
+
+    def get_editor_permissions(self):
+        permission_fields = {x: True for x in AnalysisFrameworkRole.PERMISSION_FIELDS}
+        permission_fields['can_add_user'] = False
```

Comment:
Please use a static variable for `can_add_user` and similar for others.

Reply:
Fixed